FreydCategoriesForCAP/gap/FreydCategory.gi

```diff
@@ -1742,9 +1742,247 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_FREYD_CATEGORY,
           );
           
       end );
-      
+     
     fi;
     
+    ## Random Methods
+    ##
+    if ForAll(
+          [ "RandomObjectByList",
+            "RandomObjectByInteger",
+            "RandomMorphismWithFixedSourceByList",
+            "RandomMorphismWithFixedSourceByInteger",
+            "RandomMorphismWithFixedRangeByList",
+            "RandomMorphismWithFixedRangeByInteger",
+            "RandomMorphismWithFixedSourceAndRangeByList",
+            "RandomMorphismWithFixedSourceAndRangeByInteger",
+            "RandomMorphismByList",
+            "RandomMorphismByInteger",
+            "WeakCokernelProjection",
+            "WeakKernelEmbedding" ], m -> CanCompute( underlying_category, m ) ) then
+      
+      ##
+      AddRandomObjectByList( category,
+        function ( cat, L )
+          
+          return FreydCategoryObject( cat, RandomMorphismByList( underlying_category, L ) );
+          
+      end );
+      
+      ##
+      AddRandomObjectByInteger( category,
+        function ( cat, n )
+          
+          return FreydCategoryObject( cat, RandomMorphismByInteger( underlying_category, n ) );
+          
+      end );
+      
+      ##
+      AddRandomMorphismWithFixedSourceByList( category,
+        function ( cat, S, L )
+          local rho_S, rho, u, v, pair;
+          
+          rho_S := RelationMorphism( S );
+          
+          rho := RandomMorphismWithFixedSourceByList( underlying_category, Source( rho_S ), L );
+          
+          u := UniversalMorphismIntoDirectSum(
+                        underlying_category,
+                        [ Range( rho_S ), Range( rho ) ],
+                        Source( rho_S ),
+                        [ rho_S, rho ] );
+          
+          v := WeakCokernelProjection( underlying_category, u );
+          
+          v := PreCompose( v, RandomMorphismWithFixedSourceByList( underlying_category, Range( v ), L ) );
+          
+          pair := List( [ 1, 2 ], i -> PreCompose(
+                                          underlying_category,
+                                          InjectionOfCofactorOfDirectSum( underlying_category, [ Range( rho_S ), Range( rho ) ], i ),
+                                          v ) );
+          
+          return FreydCategoryMorphism( cat, S, pair[1], FreydCategoryObject( cat, pair[2] ) );
```

Comment:
You can use `WeakBiPushout` instead, which is exactly what you construct here (`WeakCokernelProjection` of  `DirectSumCodiagonalDifference`, see https://github.com/homalg-project/CAP_project/blob/e7b9050195d9cb5af072040096535f9597791a36/FreydCategoriesForCAP/gap/FreydCategoriesDerivedMethods.gi#L427-L434 )

Reply:
Nice, I did not know that exists :-)